src/common/gui/base/colorPicker/ColorPickerView.ts

```diff
@@ -255,40 +259,65 @@ export class ColorPickerView implements Component<ColorPickerAttrs> {
 	}
 
 	private renderHuePicker(onselect: ColorPickerAttrs["onselect"]): Children {
+		const a11yHueShiftStep = 5
+
 		return m(
 			".rel.overflow-hidden",
 			{
-				tabIndex: TabIndex.Default,
+				style: {
+					position: "relative",
+				},
 				onkeydown: (e: KeyboardEvent) => {
 					e.preventDefault()
 					const isRightMove = isKeyPressed(e.key, Keys.RIGHT)
 					const isLeftMove = isKeyPressed(e.key, Keys.LEFT)
 					const isStill = isLeftMove && isRightMove
 
 					if (!isStill && (isRightMove || isLeftMove)) {
-						const step = e.shiftKey ? 1 : 5
+						// holding down shift allows finer hue adjustment
+						const step = e.shiftKey ? 1 : a11yHueShiftStep
 						let hueStep = isLeftMove ? -step : step
 						this.selectedHueAngle = normalizeHueAngle(this.selectedHueAngle + hueStep)
+
 						this.postionSliderOnHue(assertNotNull(this.hueImgDom), assertNotNull(this.hueSliderDom))
-						assertNotNull(this.hueWindowDom).style.backgroundColor = this.model.getHueWindowColor(this.selectedHueAngle)
-						assertNotNull(this.huePickerDom).style.overflow = "visible"
+						this.toggleHueWindow(true)
 						this.generatePalette()
 
 						if (!this.isAdvanced || !isValidCSSHexColor(this.customColorHex)) {
 							onselect(assertNotNull(this.palette[this.fallbackVariantIndex], `no fallback color at ${this.fallbackVariantIndex}`))
 						}
 					}
 				},
-				onkeyup: () => {
-					if (assertNotNull(this.huePickerDom).style.overflow !== "hidden") {
-						assertNotNull(this.huePickerDom).style.overflow = "hidden"
-					}
-				},
+				onkeyup: () => this.toggleHueWindow(false),
 				oncreate: (vnode) => {
 					this.huePickerDom = vnode.dom as HTMLElement
 				},
 			},
 			[
+				// range input used to allow hue to be easily changed on a screen reader
+				m("input.fill-absolute.no-hover", {
+					type: "range",
+					min: 0,
+					max: MAX_HUE_ANGLE,
+					step: a11yHueShiftStep,
+					tabIndex: TabIndex.Default,
+					role: "slider",
+					ariaLabel: lang.get("hue_label"),
+					ariaLive: "assertive",
```

Comment:
is `ariaLive` necessary? It's not going to change unless it's changed by the user. I might misunderstand when it's needed tho.

Reply:
You're absolutely right, it isn't. I initially added it when I was experimenting with relying on touch events for screen reader support